### DISCUSSION TITLE
Overlapping data inside curses applications without any addOn

### DISCUSSION THREAD

**Discussion by shashank-gowda** [sourcelair/xterm.js#4317] [Q&A]
I had previously reported an issue #3914 and my issue was duplicated with #3878. I see #3878 fix is merged to Version 5.0

Now, I have upgraded the xterm js libraries to Version 5.0 but I still see the same issue.
FYI... I'm not using add on webgl in my application.

Original Issue: #3914

**Comment by jerch** [comment]
Your issue looks like either a terminal size issue or TERM issue to me. Plz make sure to correctly update PTY size from resizes. Whats your TERM var?
Also does it work in your standard terminal?

**Comment by shashank-gowda** [comment]
Below is my Terminal settings. I'm hitting this issue when the window size is MAX.

var term = new Terminal({
            cols: 150,
            cursorBlink: true,
            cursorStyle: "block",
            scrollback: 800,
            tabStopWidth: 8,
            convertEol: true,
            fontSize: 16
        });

Yes, this works perfectly in the standard terminal.

**Comment by jerch** [comment]
> I'm hitting this issue when the window size is MAX.

So it works correctly w'o a terminal resize? Are you sure that you update the PTY size from terminal resizes?

**Comment by shashank-gowda** [comment]
Even without resize of terminal it's not working correctly. What do you mean by updating the PTY size for terminal resize?

**Comment by jerch** [comment]
The PTY maintains a terminal size itself. This gets used for canonical termios mode to do the line wrapping and size propagation to apps (SIGWINCH). Whenever a terminal emulator changes its [cols, rows]-size, the PTY size has to be updated, otherwise the line wrapping/cursor positioning will be wrong with nasty output bugs.

**Comment by shashank-gowda** [comment]
Could please help me to update the PTY size? How to do that? What calls I need to make?

**Comment by jerch** [comment]
Well I cannot tell you that, as it depends on your server side integration and how you get a working PTY from the system. `node-pty` for example exposes a `.resize` method, where the new terminal size arguments go into. In C/linux it is `ioctl(PTY_MASTER_FD, TIOCSWINSZ, &ws);`, where `ws` is a `struct winsize` containing the new values.

**Comment by LIZIMEME** [comment]
I'm having a similar problem.   The output file directory names overlap.
![image](https://user-images.githubusercontent.com/20737396/207808485-d0cd8afc-2ac6-4fda-afe3-585e99c010ec.png)

**Comment by jerch** [comment]
@LIZIMEME If `ls` print its tabbed content too far to the right leading to wrapped lines, then your TTY/PTY col size is too big.

Again - this is nothing xterm.js has control of, you have to tell the OS PTY the correct size yourself within your backend binding.
